Gems/UiBasics/CMakeLists.txt

```diff
@@ -7,7 +7,5 @@
 #
 
 # This will export its "SourcePaths" to the generated "cmake_dependencies.<project>.assetbuilder.setreg"
-if(PAL_TRAIT_BUILD_HOST_TOOLS)
-    ly_create_alias(NAME UiBasics.Builders NAMESPACE Gem)
-    ly_create_alias(NAME UiBasics.Tools NAMESPACE Gem)
-endif()
+ly_create_alias(NAME UiBasics.Builders NAMESPACE Gem)
+ly_create_alias(NAME UiBasics.Tools NAMESPACE Gem)
```

Comment:
@tkothadev put a fix for the any AssetOnly Gem into development to work when a monolithic SDK is created in this PR" https://github.com/o3de/o3de/pull/17080/files#diff-070e0bb8a7e1253b17cdf6be6dead7ed5bf1658c329bcf867b549e5f4274f7f5

This change is good as well as it makes sure the `ly_create_alias` calls exist when this is run in monolithic build when creating the SDK, therefore allowing the the UiBasics CMakeLists.txt to be added to `o3de_subdirectories_<platform>.cmake` to include the Gem targets

Reply:
Yes, the problem I encountered in this case was _other_ gems relying on UiBasics.Builders or UiBasics.Tools targets.

Reply:
Pretty much the problem is caused by the install_common logic which generates an `CMakeLists.txt` in the install layout for the equivalent directory in the source engine based on whether an `ly_create_alias` or `ly_add_target` function has been invoked at least once in the DIRECTORY CMakeLists.txt which populates the [LY_ALL_TARGET_DIRECTORIES](https://github.com/search?q=repo%3Ao3de%2Fo3de%20LY_ALL_TARGET_DIRECTORIES&type=code) property.

We then generate a CMakeList.txt for that includes a cmake file that can run the replicated `ly_create_alias` and `ly_add_target` commands.

The problem in this case was that if a monolithic build configuration is added to SDK layout after a non-monolithic build configuration, it replaces the generated CMakeLists.txt for the UiBasics Gem with an empty copy because when the `install` target ran in Monolithic it saw no `ly_create_alias` calls.

What this meant was that using that SDK layout to configure non-monolithic was broken because the `LyShine.Tools` target is looking explicitly for the [UiBasics.Tools](https://github.com/o3de/o3de/blob/c7d07cc2ad12d2062ad21318e04669642ae753b6/Gems/LyShine/Code/CMakeLists.txt#L129) target.

Monolithic SDK usage would work fine though, because `LyShine.Tools` wouldn't be exposed as a target because Tool are excluded in monolithic due to the generation logic for the SDK platform specific cmake file to create a separate non-monolithic and monolithic cmake file that gets included based on the value of LY_MONOLITHIC_GAME at configure time
https://github.com/o3de/o3de/blob/c7d07cc2ad12d2062ad21318e04669642ae753b6/cmake/Platform/Common/Install_common.cmake#L387-L396